core/eolearn/core/utils/parsing.py

```diff
@@ -81,25 +98,35 @@ class FeatureParser:
     - For `get_renamed_features` a list of triples `(feature_type, old_name, new_name)`.
     """
 
-    def __init__(self, features: Union[dict, Sequence], allowed_feature_types: Optional[Iterable[FeatureType]] = None):
+    def __init__(
+        self,
+        features: Union[SingleFeatureSpec, MultiFeatureSpec],
+        allowed_feature_types: Union[Iterable[FeatureType], EllipsisType] = ...,
+    ):
         """
         :param features: A collection of features in one of the supported formats
         :param allowed_feature_types: Makes sure that only features of these feature types will be returned, otherwise
             an error is raised
         :raises: ValueError
         """
-        self.allowed_feature_types = set(FeatureType) if allowed_feature_types is None else set(allowed_feature_types)
+        self.allowed_feature_types = (
+            set(allowed_feature_types) if isinstance(allowed_feature_types, Iterable) else set(FeatureType)
+        )
         self._feature_specs = self._parse_features(features)
 
     def _parse_features(
-        self, features: Union[dict, Sequence]
+        self, features: Union[SingleFeatureSpec, MultiFeatureSpec]
     ) -> List[Union[Tuple[FeatureType, str, str], Tuple[FeatureType, None, None]]]:
         """This method parses and validates input, returning a list of `(ftype, old_name, new_name)` triples.
 
         Due to typing issues the all-features requests are transformed from `(ftype, ...)` to `(ftype, None, None)`.
         This is a correct schema for BBOX and TIMESTAMP while for other features this is corrected when outputting,
         either by processing the request or by substituting ellipses back (case of `get_feature_specifications`).
         """
+
+        if features is FeatureType.BBOX or features is FeatureType.TIMESTAMP:
+            return [(features, None, None)]
+
```

Comment:
Is this necessary? One could just check if `features` is not a sequence, wrap it in a set if it isn't, and proceed, since such a case is handled below. This way you wouldn't duplicate the logic from L191.

Reply:
I feel like it obscures the flow a bit more, since you now need to go through another 100 lines just to figure out what happened to your `FeatureType.BBOX`. I also don't want to cast anything non-sequency into a sequence, or the users will complain when I take that away from them.

Reply:
Yes, it does obscure it...

Actually, I dislike the fact that a single tuple-like feature and the sequence of features are treated the same upon entry. It's like I would prefer to have a Feature class for this, which has a `type` and a `name`, where the `name` can be none, and also all the other properties like `is_timeless` and so on. I think it might make typing in such cases also easier.

Was such a plan ever considered and not followed due to some issues I'm unable to see at this point?

Reply:
I guess one issue here is that it introduces complexity for the users, but perhaps the users could provide the same things and it would just be used in the core functionality as a constructor of the feature class or something.

Reply:
`eo-learn` is intended to be 'user friendly' so yeah, we want to support many easy ways for them to specify features.